### ISSUE TITLE
Type toString for both mono and polymorphic functions uses the plain-printed output

### ISSUE THREAD

**Issue by Sporarum** [lampepfl/dotty#16676]
## Compiler version

Scala 3.2.0 / 3.2.1-RC2

## Minimized example & Output

```Scala
val function: Int => Int = (x: Int) => x
val polyfunction: [T] => T => T = [U] => (x: U) => x

function // somethingLambdaSomething: scala.Function1[Int, Int]
polyfunction // <function1>: 
/* scala.Polyfunction {
 *   def apply[T >: scala.Nothing <: scala.Any](x$1: T): T
 * }
 */
```
https://scastie.scala-lang.org/9VXAH9QKTyeSYE8UYdV5eQ

## Expectation
```Scala
val function: Int => Int = (x: Int) => x
val polyfunction: [T] => T => T = [U] => (x: U) => x

function // somethingLambdaSomething: Int => Int
polyfunction // <function1>: [T] => T => T
```

It's not that bad for Function1, but for polymorphic functions, it's really hard to parse, especially to newcomers !

**Comment by Sporarum** [comment]
Hello everyone,
I'm preparing some preliminaries for the spree
For this purpose, I opened a branch on dotty staging:
[fix-i16676](https://github.com/dotty-staging/dotty/tree/fix-i16676)

**Comment by Sporarum** [comment]
The only places I have been able to replicate the issue is in scastie and in metals worksheets which both use Mdoc

**Comment by Sporarum** [comment]
Opened issue on Mdoc

**Comment by mbovel** [comment]
Removing the Spree label as @Sporarum is assigned to this issue.

**Comment by SethTisue** [comment]
I'm not able to reproduce this in any version of Scala

with the exception that 3.0.0 did print the types of polymorphic functions a bit strangely:

```scala
scala> val polyfunction: [T] => T => T = [U] => (x: U) => x                                         
val polyfunction: PolyFunction{apply: [T](x$1: T): T} = ...
```

but that already got fixed in Scala 3.1.0, and current nightly has:

```scala
scala> val polyfunction: [T] => T => T = [U] => (x: U) => x
val polyfunction: [T] => (x$1: T) => T = ...
```

(as an aside, it's odd and seems wrong to me that the synthetic parameter name `x$1` appears — it didn't in 3.2.0, but then began appearing in  3.3.0 — I can't tell if the change was intentional or not — but that would be a separate ticket)

anyway, @Sporarum , afaict this ticket should simply be closed, as it seems we don't have any definite information that there is a Scala bug here. do you agree? or if not, can you provide a current reproduction?

cc @bvenners

**Comment by Sporarum** [comment]
> The only places I have been able to replicate the issue is in scastie and in metals worksheets

So it was already working in the REPL when I opened this issue (I also remember trying it)

If you check on scastie, both the version I opened this issue for (3.2.0), and the latest version (3.7.2) have the problem as described

It is possible the issue doesn't come from the compiler, but we need to find where before I feel confident in closing this issue
